Source/OCMock/NSInvocation+OCMAdditions.m

```diff
@@ -55,12 +55,11 @@ + (NSInvocation *)invocationForBlock:(id)block withArguments:(NSArray *)argument
 
 static NSString *const OCMRetainedObjectArgumentsKey = @"OCMRetainedObjectArgumentsKey";
 
-- (void)retainObjectArgumentsExcludingObject:(id)objectToExclude
+- (void)retainObjectArgumentsExcludingObject:(id)objectToExclude excludingObjectsAtIndexes:(NSIndexSet *)indexes;
 {
-    if(objc_getAssociatedObject(self, OCMRetainedObjectArgumentsKey) != nil)
+    if (objc_getAssociatedObject(self, OCMRetainedObjectArgumentsKey) != nil)
     {
-        // looks like we've retained the arguments already; do nothing else
-        return;
+        [NSException raise:NSInternalInconsistencyException format:@"Arguments have already been retained"];
```

Comment:
Why is this important? I'm really worried about other codebases. While I understand your intent, as you are improving a large codebase, I don't want OCMock 3.7 to be the one that users hate because it breaks their builds. At this stage stricter checks should probably be opt-in, right?

Reply:
Removed for the time being.